src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -533,21 +533,66 @@ case object MSTORE8 extends OpCode(0x53, 2, 0, _.G_verylow) {
 
 case object SSTORE extends OpCode(0x55, 2, 0, _.G_zero) {
   protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
-    val (Seq(offset, value), stack1) = state.stack.pop(2)
-    val oldValue = state.storage.load(offset)
-    val refund: BigInt = if (value.isZero && !UInt256(oldValue).isZero) state.config.feeSchedule.R_sclear else 0
-    val updatedStorage = state.storage.store(offset, value)
+    val (Seq(offset, newValue), stack1) = state.stack.pop(2)
+    val currentValue = state.storage.load(offset)
+    val refund: BigInt = if(isAfterConstantinopleFork(state)){
```

Comment:
Maybe create separate functions for pre-consatninople and post-constatninople rules ?